### PR TITLE
Query OSM and import data to osm-p2p

### PR DESCRIPTION
**Pull request by dereklieu**
- [x] translates bounding box to osm.org query
- [x] queries osm.org and returns xml
- [x] imports xml to osm-p2p

@noffle I could use a set of eyes on the last step. Right now `osm-p2p-db-importer` is throwing an error on [`collect()`](https://github.com/noffle/osm-p2p-db-importer/blob/master/index.js#L29), and I have no idea why.

In addition, the process of [closing the osm db, removing the data, then re-initiating the db](https://github.com/osmlab/field-data-coordinator/pull/16/files#diff-be228df81b99a384aa50ad50bde7fca0R45) is a bit awkward, and I wonder if there's a better way to do that.

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -37,6 +42,20 @@ function createObservationsReplicationStream () {
   return observationsDb.log.replicate()
 }
 
+function importBulkOsm (xmlReadableStream, cb) {
+  if (osmOrgDb) { osmOrgDb.db.close() }
```
Comment (noffle) [inline lib/db.js]: `close()` is an async method -- you might need to pass it a callback and wait for it to return before doing the rimraffing below.

File: src/sagas/index.js [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+  all
+} = require('redux-saga/effects')
+
+const {
+  listObservations,
+  importSurvey,
+  importOsm
+} = require('../drivers/local')
+
+const OSMAPI = 'http://api.openstreetmap.org/api/0.6/'
```
Comment (mojodna) [inline src/sagas/index.js]: Let's make this configurable earlier-on (potentially via environment var for now?).

File: lib/db.js [hunk 3]
```diff
@@ -44,6 +47,10 @@ function createObservationsReplicationStream () {
   return observationsDb.log.replicate()
 }
 
+function wipeDb (db, path, cb) {
+  db.close(() => rimraf(path, () => mkdirp(path, cb)))
```
Comment (noffle) [inline lib/db.js]: If `rimraf` fails, its error will be lost.

Comment (noffle) [inline lib/db.js]: Same with db.close()'s error.

File: lib/db.js [hunk 4]
```diff
@@ -54,7 +61,7 @@ function importBulkOsm (bbox, cb) {
     })
   }
   if (osmOrgDb) {
-    osmOrgDb.db.close(() => {
+    wipeDb(osmOrgDb.db, osmOrgDbPath, () => {
```
Comment (noffle) [inline lib/db.js]: Do we care about error handling here?

Comment (dereklieu) [inline lib/db.js]: Yeah we definitely do. This should be addressed in [1f809d7](https://github.com/osmlab/field-data-coordinator/pull/16/commits/1f809d715b6287297128a9ed2a4cfcfe6ddc94a8)

**Pull request merged**